version: '3.9'

services:
  django-app:
    build:
      context: .
      dockerfile: deploy/Dockerfile.django.prod
      network: host
    ports:
      - "8000:8000"
    env_file:
      - .env
    environment:
      ENVIRONMENT: PROD
    volumes:
      - .:/app
      - /var/www/static:/app/static
    depends_on:
      db:
        condition: service_healthy

  db:
    image: postgres:latest
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_DATABASE}
    volumes:
      - postgres-data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    env_file:
      - .env
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER} && PGPASSWORD=${DB_PASSWORD} psql -U ${DB_USER} -d ${DB_DATABASE} -c 'SELECT 1'"]
      interval: 10s
      timeout: 3s
      retries: 3

  bot:
    build:
      context: .
      dockerfile: deploy/Dockerfile.bot
      network: host
    volumes:
      - ./bot:/app/bot
    env_file:
      - .env
    restart: always

volumes:
  postgres-data:
